server/coordinator/procedure/create_table.go

```diff
@@ -0,0 +1,168 @@
+// Copyright 2022 CeresDB Project Authors. Licensed under Apache-2.0.
+
+package procedure
+
+import (
+	"context"
+	"sync"
+
+	"github.com/CeresDB/ceresdbproto/pkg/metaservicepb"
+	"github.com/CeresDB/ceresmeta/server/cluster"
+	"github.com/CeresDB/ceresmeta/server/coordinator/eventdispatch"
+	"github.com/looplab/fsm"
+	"github.com/pkg/errors"
+)
+
+const (
+	eventCreateTablePrepare = "EventCreateTablePrepare"
+	eventCreateTableFailed  = "EventCreateTableFailed"
+	eventCreateTableSuccess = "EventCreateTableSuccess"
+
+	stateCreateTableBegin   = "StateCreateTableBegin"
+	stateCreateTableWaiting = "StateCreateTableWaiting"
+	stateCreateTableFinish  = "StateCreateTableFinish"
+	stateCreateTableFailed  = "StateCreateTableFailed"
+)
+
+var (
+	createTableEvents = fsm.Events{
+		{Name: eventCreateTablePrepare, Src: []string{stateCreateTableBegin}, Dst: stateCreateTableWaiting},
+		{Name: eventCreateTableSuccess, Src: []string{stateCreateTableWaiting}, Dst: stateCreateTableFinish},
+		{Name: eventCreateTableFailed, Src: []string{stateCreateTableWaiting}, Dst: stateCreateTableFailed},
+	}
+	createTableCallbacks = fsm.Callbacks{
+		eventCreateTablePrepare: createTablePrepareCallback,
+		eventCreateTableFailed:  createTableFailedCallback,
+		eventCreateTableSuccess: createTableSuccessCallback,
+	}
+)
+
+func createTablePrepareCallback(event *fsm.Event) {
+	request := event.Args[0].(*CreateTableCallbackRequest)
+	table, exists, err := request.cluster.GetOrCreateTable(request.ctx, request.nodeName, request.schemaName, request.tableName)
+	if err != nil {
+		cancelEventWithLog(event, err, "cluster get or create table")
+		return
+	}
+	if exists {
+		return
+	}
+
+	shard, err := request.cluster.GetShardByID(table.GetShardID())
+	if err != nil {
+		cancelEventWithLog(event, err, "cluster get shard by id")
+		return
+	}
+	// TODO: consider followers
+	leader := shard.GetLeader()
+	if leader == nil {
+		cancelEventWithLog(event, ErrShardLeaderNotFound, "shard can't find leader")
+		return
+	}
+
+	err = request.dispatch.CreateTableOnShard(request.ctx, leader.Node, &eventdispatch.CreateTableOnShardRequest{
+		TableInfo: &cluster.TableInfo{
+			ID:         table.GetID(),
+			Name:       table.GetName(),
+			SchemaID:   table.GetSchemaID(),
+			SchemaName: table.GetSchemaName(),
+		},
+		CreateSQL: request.createSQL,
```

Comment:
The protocol has been updated by https://github.com/CeresDB/ceresdbproto/pull/27, the sql is not enough now...

Reply:
I will update later after this pr is merged.